code/modules/admin/verbs/playsound.dm

```diff
@@ -26,14 +26,24 @@ GLOBAL_LIST_EMPTY(sounds_cache)
 	if(alert("Are you sure?\nSong: [S]\nNow you can also play this sound using \"Play Server Sound\".", "Confirmation request" ,"Play", "Cancel") == "Cancel")
 		return
 
+	if(holder.fakekey)
+		if(alert("Playing this sound will expose your real ckey despite being in stealth mode. You sure?", "Double check" ,"Play", "Cancel") == "Cancel")
+			return
+
+
 	log_admin("[key_name(src)] played sound [S]")
 	message_admins("[key_name_admin(src)] played sound [S]", 1)
 
 	for(var/mob/M in GLOB.player_list)
 		if(M.client.prefs.sound & SOUND_MIDI)
+			if(ckey in M.client.prefs.admin_sound_ckey_ignore)
+				continue // This player has this admin muted
```

Comment:
If you are doing this, please also make it tie into local sounds, not just global ones.

Reply:
Not so easy, that hooks into `playsound()` which doesnt let me iterate players easily outside of it, so I would have to dupe half the code. It is still doable though.

Reply:
![image](https://user-images.githubusercontent.com/25063394/223246443-8aebcda3-9b4e-4a9a-9664-deaf8dbac8a1.png)